### ISSUE TITLE
Example in asynchttpserver pegs CPU at 100% and hogs memory

### ISSUE THREAD

**Issue by PMunch** [nim-lang/Nim#17783]
[Brought up by mrotaru on Discord](https://irclogs.nim-lang.org/19-04-2021.html#13:28:18), the asynchttpserver example doesn't work on any Nim version that can run it. Tested on Manjaro and Ubuntu, and both peg a CPU core at 100% and starts eating memory until the machine runs out. All the while not replying to requests. This is similar to the behaviour I saw in my HomeAssistant add-ons after turning my callbacks async, so this might be related to the async callback in this code.

**Comment by planetis-m** [comment]
Duplicate #16506

**Comment by dom96** [comment]
What stack trace is the code getting stuck on?

**Comment by PMunch** [comment]
```nim
Traceback (most recent call last)
/tmp/test.nim(17)        test
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/pure/asyncmacro.nim(262) main
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/pure/asyncmacro.nim(29) mainNimAsyncContinue
/tmp/test.nim(13)        mainIter
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/pure/asyncmacro.nim(262) acceptRequest
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/pure/asyncmacro.nim(29) acceptRequestNimAsyncContinue
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/pure/asynchttpserver.nim(339) acceptRequestIter
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/pure/asyncnet.nim(476) acceptAddr
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/system.nim(240) acceptAddr
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/system/gc.nim(439) newObj
/home/peter/.choosenim/toolchains/nim-1.4.6/lib/system/cellseqs_v1.nim add
SIGINT: Interrupted by Ctrl-C.
```

`server.shouldAcceptRequest()` also seems to always be true by the way.

**Comment by dom96** [comment]
This was fixed in https://github.com/nim-lang/Nim/pull/16580. Hasn't made it to a released version yet though... so the docs at https://nim-lang.org/docs/asynchttpserver.html are wrong :/

**Comment by Araq** [comment]
What? This should have been backported ... Ping @narimiran

**Comment by narimiran** [comment]
> This was fixed in #16580. 

> This should have been backported

The fix is now backported to the v1.4 branch and it will be part of the next 1.4.x release.
